v1.0.0-rc.6 release (#6314)
ZcashFoundation/zebra: CHANGELOG.md

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org).

## [Zebra 1.0.0-rc.6](https://github.com/ZcashFoundation/zebra/releases/tag/v1.0.0-rc.6) - 2023-03-23

> In this release, we fixed several security issues, most notably the [vulnerabilities recently disclosed by Halborn](https://zfnd.org/statement-on-recent-security-disclosures-by-halborn/).

Reviewer: It's important that we distinguish between the major Bitcoin fork vulnerabilities, and the minor similar issues in our code.

```suggestion
In this release, we fixed several minor security issues, most notably [hardening Zebra in response to the vulnerabilities recently disclosed by Halborn](https://zfnd.org/statement-on-recent-security-disclosures-by-halborn/).
```

> In this release, we fixed several security issues, most notably the [vulnerabilities recently disclosed by Halborn].(https://zfnd.org/statement-on-recent-security-disclosures-by-halborn/).

Reviewer: I think it needs to be like this to render the link

```suggestion
In this release, we fixed several security issues, most notably the [vulnerabilities recently disclosed by Halborn](https://zfnd.org/statement-on-recent-security-disclosures-by-halborn/).
```

### Known Issues

> - `orchard` 0.3.0 can't verify `halo2` proofs when compiled with Rust 1.69 or later (currently beta Rust). Compile Zebra with stable Rust to avoid this bug. ([halo2/#737](https://github.com/zcash/halo2/issues/737)).

Reviewer: ```suggestion
- `orchard` 0.3.0 can't verify `halo2` proofs when compiled with Rust 1.69 or later (currently beta and nightly Rust). Compile Zebra with stable Rust to avoid this bug. ([halo2/#737](https://github.com/zcash/halo2/issues/737)).
```

Reviewer 2: Thanks, changed this at https://github.com/ZcashFoundation/zebra/pull/6314/commits/845dea26ca81ad7429990446b5232fa9b0dbdb68

Zebra tracking issue for this problem is [here](https://github.com/ZcashFoundation/zebra/issues/6232).

### Security

> - Fix Halborn disclosures in Zebra ([#6297](https://github.com/ZcashFoundation/zebra/pull/6297))

Reviewer: ```suggestion
- Harden Zebra's network protocol implementation in response to the Holborn disclosures ([#6297](https://github.com/ZcashFoundation/zebra/pull/6297))
```

Reviewer 2: https://github.com/ZcashFoundation/zebra/pull/6314/commits/845dea26ca81ad7429990446b5232fa9b0dbdb68

> - Fix Halborn disclosures in Zebra ([#6297](https://github.com/ZcashFoundation/zebra/pull/6297))

Reviewer: Do you think that the Halborn disclosures should be in the release summary?
We could use some of the descriptions from the blog post, and also link to it so people can find more information:
https://zfnd.org/statement-on-recent-security-disclosures-by-halborn/

Reviewer 2: Added to the summary at https://github.com/ZcashFoundation/zebra/pull/6314/commits/845dea26ca81ad7429990446b5232fa9b0dbdb68

> Zebra tracking issue for this problem is [here](https://github.com/ZcashFoundation/zebra/issues/6232).

Reviewer: ```suggestion
Zebra tracking issue for this problem is [#6232](https://github.com/ZcashFoundation/zebra/issues/6232).
```

> - Bump incrementalmerkletree from 0.3.0 to 0.3.1, resolving a consensus bug on 32-bit platforms ([#6258](https://github.com/ZcashFoundation/zebra/pull/6258))

Reviewer: Optional: Do you think that the 32-bit platform incompatibility should be in the release summary? It seems like it might be important to some users.

 But we don't support 32-but platforms, so maybe that is ok where it is.

Reviewer 2: Lets leave it there.

> - Harden Zebra's network protocol implementation in response to the Holborn disclosures ([#6297](https://github.com/ZcashFoundation/zebra/pull/6297))

Author: ```suggestion
- Harden Zebra's network protocol implementation in response to the Halborn disclosures ([#6297](https://github.com/ZcashFoundation/zebra/pull/6297))
```

Reviewer 2: Thanks, fixed at https://github.com/ZcashFoundation/zebra/pull/6314/commits/59dc1976bd9d75fa8026de10afe84179615f0e6d

- Remove unused dependencies, and check for them in CI ([#6216](https://github.com/ZcashFoundation/zebra/pull/6216))

> - Validate address length before reading ([#6320](https://github.com/ZcashFoundation/zebra/pull/6320)) [#6368](https://github.com/ZcashFoundation/zebra/pull/6368)

Reviewer: ```suggestion
- Validate address length before reading ([#6320](https://github.com/ZcashFoundation/zebra/pull/6320), [#6368](https://github.com/ZcashFoundation/zebra/pull/6368))
```
